account_payment_return/models/payment_return.py

```diff
@@ -0,0 +1,193 @@
+        invoices = self.env['account.invoice'].search(
+            [('move_id', 'in', invoice_moves.ids)])
+        return invoices
+
+    @api.one
+    def action_confirm(self):
+        move_obj = self.env['account.move']
+        # Check for incomplete lines
+        for return_line in self.line_ids:
+            if not return_line.move_line_id:
+                raise except_orm(_('Error!'),
+                                 _("You must complete all moves "
+                                   "references in the "
+                                   "payment return."))
+
+        move = {
+            'name': '/',
+            'ref': _('Return %s') % self.name,
+            'journal_id': self.journal_id.id,
+            'date': self.date,
+            'company_id': self.company_id.id,
+        }
+        # Period
+        if self.period_id:
+            move['period_id'] = self.period_id.id
+        else:
+            context_move = self._context.copy()
+            context_move.update({'company_id':
+                                 self.company_id.id})
+            move['period_id'] = self.period_id.with_context(
+                context_move).find(self.date).id
+
+        move_id = move_obj.create(move)
+
+        for return_line in self.line_ids:
+            move_line = return_line.move_line_id
+            if move_line:
+                old_reconcile = move_line.reconcile_id
+                lines2reconcile = old_reconcile.line_id
+                invoices = self._get_invoices(lines2reconcile)
+                move_line_id = move_line.copy(
+                    default={
+                        'move_id': move_id.id,
+                        'ref': move['ref'],
+                        'date': move['date'],
+                        'period_id': move['period_id'],
+                        'journal_id': move['journal_id'],
+                        'debit': return_line.amount,
+                        'credit': 0,
+                    })
+                lines2reconcile += move_line_id
+                move_line_id.copy(
+                    default={
+                        'debit': 0,
+                        'credit': return_line.amount,
+                        'account_id': move_line.move_id.journal_id
+                        .default_credit_account_id.id,
+                    })
+            # Break old reconcile and
+            # make a new one with at least three moves
+            old_reconcile.unlink()
+            lines2reconcile.reconcile_partial()
+            move_line = move_line_id
+            return_line.write({'reconcile_id': move_line
+                               .reconcile_partial_id.id})
+            # Mark invoice as payment refused
+            invoices.write({'payment_returned': True})
+
+        move_id.button_validate()
+        self.write({'state': 'done', 'move_id': move_id.id})
+        return True
+
+    @api.one
+    def action_cancel(self):
+        if self.move_id:
```

Comment:
IMO you can reduce the number of nested python blocks and improve the readability with:
```python
if not self.move_id:
    return True
for return_line in self.line_ids:
    ....
self.write(....
return True
```